src/engine/meta/meta_blks_mgr.cpp

```diff
@@ -1435,9 +1447,90 @@ bool MetaBlkMgr::sanity_check(const bool check_ovf_chain) {
 //
 sisl::status_response MetaBlkMgr::get_status(const sisl::status_request& request) {
     sisl::status_response response;
-    std::string dummy_client;
-    response.json = populate_json(request.verbose_level, m_meta_blks, m_ovf_blk_hdrs, m_last_mblk_id.get(), m_sub_info,
-                                  m_self_recover, dummy_client);
+    nlohmann::json j;
+    {
+        std::lock_guard< decltype(m_meta_mtx) > lg{m_meta_mtx};
+        j["ssb"] = m_ssb ? m_ssb->to_string() : "";
+        j["self_recovery"] = m_self_recover;
+        j["last_mid"] = m_last_mblk_id->to_string();
+        j["compression"] = compress_feature_on() ? "On" : "Off";
+    }
+
+    response.json = j;
+    return response;
+}
+sisl::status_response MetaBlkMgr::get_status_metablk(const sisl::status_request& request, std::string child_type) {
+
+    auto log_level{request.verbose_level};
+    sisl::status_response response;
+    std::string metablk_type = request.json.contains("name") ? request.obj_name : child_type;
+
+    nlohmann::json& j = response.json;
+    {
+        std::lock_guard< decltype(m_meta_mtx) > lg{m_meta_mtx};
+        std::string metablk_name_template{"MetaBlk_"};
+        auto pos{metablk_name_template.size()};
+        auto client = metablk_type.substr(pos);
+        if (m_sub_info.find(client) == m_sub_info.end()) {
+            j["error"] = "no client " + client + "  found";
+            return response;
+        }
+        for (auto& x : m_sub_info) {
+            if (!client.empty() && client.compare(x.first)) { continue; }
+
+            j[x.first]["type"] = x.first;
+            j[x.first]["do_crc"] = x.second.do_crc;
+            j[x.first]["cb"] = x.second.cb ? "registered valid cb" : "nullptr";
+            j[x.first]["comp_cb"] = x.second.comp_cb ? "registered valid cb" : "nullptr";
+            j[x.first]["num_meta_bids"] = x.second.meta_bids.size();
+            if (log_level >= 2 && log_level <= 3) {
+                size_t bid_cnt{0};
+                uint32_t start_sub_type = request.json.contains("start_sub_type_id")
+                    ? std::stoul(request.json["start_sub_type_id"].get< std::string >())
+                    : 0;
+                uint32_t end_sub_type = request.json.contains("end_sub_type_id")
+                    ? std::stoul(request.json["end_sub_type_id"].get< std::string >())
+                    : UINT32_MAX;
+
+                for (const auto& y : x.second.meta_bids) {
+                    if (bid_cnt < start_sub_type || bid_cnt > end_sub_type) {
+                        bid_cnt++;
+                        continue;
+                    }
+                    BlkId bid(y);
+                    auto it = m_meta_blks.find(y);
+                    std::string jname = "content";
+                    sisl::byte_array buf;
+                    if (client == "VOLUME" || log_level == 3) {
+                        HS_DBG_ASSERT_EQ(it != m_meta_blks.end(), true,
```

Comment:
Remove all ASSERT and convert them to error response to http request. We can't crash in produce by running http request.